From c710f4845afe99f6f2f9ae8594c92c2fcd4f15da Mon Sep 17 00:00:00 2001
From: James Lu <james@overdrivenetworks.com>
Date: Sun, 24 Dec 2017 11:05:21 -0800
Subject: [PATCH 2/4] backend/gstreamer: record sound in stereo

Bug: https://bugs.launchpad.net/kazam/+bug/1179073
---
 kazam/backend/gstreamer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kazam/backend/gstreamer.py b/kazam/backend/gstreamer.py
index 8e35db7..0729e59 100644
--- a/kazam/backend/gstreamer.py
+++ b/kazam/backend/gstreamer.py
@@ -234,7 +234,7 @@ class Screencast(GObject.GObject):
             logger.debug("Audio1 Source:\n  {0}".format(self.audio_source))
             self.audiosrc = Gst.ElementFactory.make("pulsesrc", "audio_src")
             self.audiosrc.set_property("device", self.audio_source)
-            self.aud_caps = Gst.caps_from_string("audio/x-raw")
+            self.aud_caps = Gst.caps_from_string("audio/x-raw,channels=2")
             self.aud_caps_filter = Gst.ElementFactory.make("capsfilter", "aud_filter")
             self.aud_caps_filter.set_property("caps", self.aud_caps)
 
@@ -244,7 +244,7 @@ class Screencast(GObject.GObject):
             logger.debug("Audio2 Source:\n  {0}".format(self.audio2_source))
             self.audio2src = Gst.ElementFactory.make("pulsesrc", "audio2_src")
             self.audio2src.set_property("device", self.audio2_source)
-            self.aud2_caps = Gst.caps_from_string("audio/x-raw")
+            self.aud2_caps = Gst.caps_from_string("audio/x-raw,channels=2")
             self.aud2_caps_filter = Gst.ElementFactory.make("capsfilter", "aud2_filter")
             self.aud2_caps_filter.set_property("caps", self.aud2_caps)
             self.aud2_in_queue = Gst.ElementFactory.make("queue", "queue_a2_in")
-- 
2.15.1

